jetty-core/jetty-http-spi/src/main/java/org/eclipse/jetty/http/spi/JettyHttpExchangeDelegate.java

```diff
@@ -68,10 +69,40 @@ public Headers getRequestHeaders()
 
         for (HttpField field : _request.getHeaders())
         {
-            if (field.getValue() == null)
+            String rawValue = field.getValue();
+            if (rawValue == null)
                 continue;
-            for (String value : field.getValues())
-                headers.add(field.getName(), value);
+
+            String headerName = field.getLowerCaseName();
```

Comment:
No, use `field.getHeader()` so there is no need to lowercase and the switch will be more efficient.

Reply:
`field.getHeader()` returns an `HttpHeader`, which is null for headers that are not hardcoded in the `HttpHeader` class.